### PR TITLE
API extensions

### PR DESCRIPTION
**Pull request by kiliit**
Some extensions of the api

### CODE REVIEW

File: repo_commits.go [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+	Addition		   int 					`json:"addition"`
+	Deletion 		   int					`json:"deletion"`
+	Type               string				`json:"type"`
+	IsCreated          bool					`json:"created"`
+	IsDeleted          bool					`json:"deleted"`
+	IsBin              bool					`json:"bin"`
+	Sections           []*DiffSection		`json:"sections"`
+}
+
+type Diff struct {
```
Comment (Unknwon) [inline repo_commits.go]: What does this struct do? Seems not using is anywhere?

Comment (kiliit) [inline repo_commits.go]: Oh. I didn't create the client method in the gogs client.

The DiffFile struct is used in gogs. 

I tested the service method from the java lib and forgot the client method in gogs-client

Comment (Unknwon) [inline repo_commits.go]: OK... should you remove it or...?

Comment (kiliit) [inline repo_commits.go]: No. It is used by Gogs, and is needed for my gogs pull request. See the DiffRange function in the that pull request. 

This is also why the gogs pull request has dependency to kiliit/go-gogs-client. If it hasn't then compile fails.

Comment (Unknwon) [inline repo_commits.go]: Sorry I don't get it...

Gogs has these strcut itself, why need import from this package?

**Comment by Unknwon** [comment]
Hi, can you resolve conflicts?

**Pull request closed without merging**